src/components/adminDashboard/components/eMetadata/services/eMetadata.service.js

```diff
@@ -0,0 +1,121 @@
+(function (angular) {
+    'use strict';
+
+    angular.module('znk.infra-web-app.adminDashboard')
+        .service('EMetadataService',
+            function ($mdDialog, $http, ENV, $q, InfraConfigSrv, $log) {
+                'ngInject';
+
+
+                var self = this;
+                  var profilePath = ENV.backendEndpoint + "/teachworks/zinkerzTeacher/all";
+             //   var profilePath = "http://localhost:3009/teachworks/zinkerzTeacher/all";
+                var globalStorageProm = InfraConfigSrv.getGlobalStorage();
+
+                var satURL = "https://sat-dev.firebaseio.com";
+                var actURL = "https://act-dev.firebaseio.com";
+                var tofelURL = "https://znk-toefl-dev.firebaseio.com";
+
+                if (!ENV.debug) {
+                    // satURL = "https://sat-dev.firebaseio.com/";
+                    // actURL = "https://act-dev.firebaseio.com/";
+                    // tofelURL = "https://znk-toefl-dev.firebaseio.com/";
+                }
+
+
+                self.showEducatorProfile = function (userProfile) {
+                    $q.all([getTimezonesList(), getLocalTimezone()]).then(function (values) {
+                        var timezonesList = values[0];
+                        var localTimezone = values[1];
+                        $mdDialog.show({
+                            locals: {
+                                userProfile: userProfile,
+                                timezonesList: obj2Array(timezonesList),
+                                localTimezone: localTimezone
+                            },
+                            controller: 'EducatorProfileController',
+                            controllerAs: 'vm',
+                            templateUrl: 'components/adminDashboard/components/eMetadata/templates/educatorProfile.template.html',
+                            clickOutsideToClose: true,
+                            escapeToClose: true
+                        });
+                    });
+
+                };
+
+                self.updateProfile = function (newProfile) {
+                    var copiedProfile =angular.copy(newProfile);
+                    var uid = copiedProfile.uid;
+                    if (uid) {
+                        delete copiedProfile.uid;
+                    }
+                    var fullPath = "users/" + uid + "/profile";
+                    return InfraConfigSrv.getGlobalStorage().then(function (globalStorage) {
+                        return globalStorage.update(fullPath, copiedProfile);
```

Comment:
do this function has to return a promise? who waits for it? what will happen in case of an error